mars/services/scheduling/supervisor/manager.py

```diff
@@ -86,6 +102,14 @@ async def __post_create__(self):
         self._global_slot_ref = await mo.actor_ref(
             GlobalSlotManagerActor.default_uid(), address=self.address
         )
+        self._speculation_execution_scheduler = SpeculativeScheduler(
+            self._queueing_ref, self._global_slot_ref, self._speculation_config
+        )
+        await self._speculation_execution_scheduler.start()
+
+    async def __pre_destroy__(self):
+        if self._subtask_speculation_enabled:
```

Comment:
Why `if self._subtask_speculation_enabled`? If `self._speculation_execution_scheduler` exists, we should stop it.